### PR TITLE
PowerISO

### PR DESCRIPTION
**Pull request by hadess**
PowerISO is freeware, and I was allowed to "use [it] or distribute
it for free."

### CODE REVIEW

**Comment by hadess** [comment]
I'll add mime-types as soon as I've figured out what it supports ;)

File: com.poweriso.PowerISO.json [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+                {
+                    "type": "archive",
+                    "url": "https://www.poweriso.com/poweriso-1.5.tar.gz",
+                    "sha256": "1893189e0c8b77da179df64a64cf8e62c4216684d1ab262867c6c96faae64ee2",
+                    "strip-components": 0,
+                    "dest": "cli"
+                },
+                {
+                    "type": "file",
+                    "path": "com.poweriso.PowerISO.appdata.xml"
```
Comment (LeandroStanger) [inline com.poweriso.PowerISO.json:43]: New appdata files should use metainfo instead. https://www.freedesktop.org/software/appstream/docs/chap-Quickstart.html

**Comment by nedrichards** [comment]
bot, build com.poweriso.PowerISO

File: com.poweriso.PowerISO.desktop [hunk 2]
```diff
@@ -0,0 +1,10 @@
+[Desktop Entry]
+Name=PowerISO
+Comment=CD/DVD/Blu-Ray processing tool
+Type=Application
+Categories=Utility;DiscBurning;AudioVideo;
+Exec=/app/bin/PowerISO %F
```
Comment (eszlari) [inline com.poweriso.PowerISO.desktop:6]: path could be dropped

File: launcher.sh [hunk 3]
```diff
@@ -0,0 +1,10 @@
+#!/bin/sh
+
+OLDDIR=`pwd`
+LIBDIR=/app/lib/poweriso
+export LD_LIBRARY_PATH=$LIBDIR
+export QT_QPA_PLATFORM_PLUGIN_PATH=$LIBDIR
+# export QT_DEBUG_PLUGINS=1
+cd $LIBDIR
+./poweriso
```
Comment (eszlari) [inline launcher.sh:9]: `exec ./poweriso "$@"`

File: com.poweriso.PowerISO.json [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    ],
+    "modules": [
+        {
+            "name": "poweriso",
+            "buildsystem": "simple",
+            "build-commands": [
+                "mkdir /app/lib/",
+                "cp -r ./gui/ /app/lib/poweriso/",
+                "mkdir -p /app/bin/",
+                "install -m755 launcher.sh /app/bin/PowerISO",
```
Comment (eszlari) [inline com.poweriso.PowerISO.json:21]: Maybe `poweriso-gui` would be more descriptive.

File: launcher.sh (hunk 3 above)
Comment (hadess) [reply launcher.sh]: It doesn't actually handle any command-line args.

File: com.poweriso.PowerISO.desktop (hunk 2 above)
Comment (hadess) [reply com.poweriso.PowerISO.desktop]: Done.

**Comment by gasinvein** [comment]
Doesn't it require root privileges or fuse to mount images?

**Comment by hadess** [comment]
> Doesn't it require root privileges or fuse to mount images?

PowerISO doesn't "mount" images.

**Comment by nedrichards** [comment]
bot, build com.poweriso.PowerISO

File: com.poweriso.PowerISO.json [hunk 5]
```diff
@@ -0,0 +1,60 @@
+{
+    "app-id": "com.poweriso.PowerISO",
+    "runtime": "org.freedesktop.Platform",
+    "runtime-version": "20.08",
+    "sdk": "org.freedesktop.Sdk",
+    "command": "poweriso-gui",
+    "finish-args": [
+        "--device=all",
+        "--socket=x11",
+        "--socket=wayland",
```
Comment (barthalion) [inline com.poweriso.PowerISO.json:10]: Does it actually work under Wayland or fallbacks to xwayland?

**Pull request closed without merging**